File: Screamer.Domain/Entities/Reaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Screamer.Domain.Common;
using Screamer.Identity.Models;

namespace Screamer.Domain.Entities
{
    public class Reaction : BaseEntity
    {
        public string UserId { get; set; }
        [ForeignKey("UserId")]
        public ApplicationUser User { get; set; }
        public int PostId { get; set; }
        [Required]
        [ForeignKey("PostId")]
        public Post Post { get; set; }
        [Required]
        [ForeignKey("CommentId")]
        public Comment Comment { get; set; }

        public int CommentId { get; set; }

        public string ReactionType { get; set; }
    }
}

File: Screamer.AngularUI/src/app/core/models/Reaction.ts
import { BaseModel } from "./BaseModel"
import { Post } from "./Post";
import { User } from "./User";


export interface Reaction extends BaseModel {
        Name: string;
        Color: string;
        Value: string;
        Icon: string;
        PostId: string;
        CommentId: string;
        UserId: number;
        User: User;
        Post: Post;
        Comment: Comment;
    }
